evalml/pipelines/components/transformers/preprocessing/delayed_feature_transformer.py

```diff
@@ -74,11 +77,7 @@ def _encode_y_while_preserving_index(y):
 
     @staticmethod
     def _get_categorical_columns(X):
-        return [
-            name
-            for name, column in X.ww.columns.items()
-            if isinstance(column.logical_type, logical_types.Categorical)
-        ]
+        return list(X.ww.select(["categorical"], return_schema=True).columns)
```

Comment:
Not related to this issue but figured we could simplify the existing implementation with `return_schema`